pydbgen/pydbgen.py

```diff
@@ -45,266 +69,151 @@ def license_plate(self,seed=None,style=None):
 
+        return license_place_format.format(p1=p1, p2=p2, p3=p3)
 
-    def realistic_email(self,name,seed=None):
+
+    def realistic_email(self, name, seed=None):
         '''
         Generates realistic email from first and last name and a random domain address
         seed: Currently not used. Uses seed from the pydb class if chosen by user
         '''
-        import random
-        import os
-        from random import randint,choice
         random.seed(self.seed)
 
-        name=str(name)
-        result=''
+        name = str(name)
         f_name = name.split()[0]
         l_name = name.split()[-1]
-        
+
         choice_int = choice(range(10))
-        
-         
-        dir_path = os.path.dirname(os.path.realpath(__file__))
-        print(path)
-        path = dir_path+"\Domains.txt"
-        
-        domain_list = []
-        fh = open(path)
-        for line in fh.readlines():
-            domain_list.append(str(line).strip())
-
-        domain = choice(domain_list)
-        fh.close()       
-        
-        name_choice = choice(range(8))
-        
-        if name_choice==0:
-            name_combo=f_name[0]+l_name
-        elif name_choice==1:
-            name_combo=f_name+l_name
-        elif name_choice==2:
-            name_combo=f_name+'.'+l_name[0]
-        elif name_choice==3:
-            name_combo=f_name+'_'+l_name[0]
-        elif name_choice==4:
-            name_combo=f_name+'.'+l_name
-        elif name_choice==5:
-            name_combo=f_name+'_'+l_name
-        elif name_choice==6:
-            name_combo=l_name+'_'+f_name
-        elif name_choice==7:
-            name_combo=l_name+'.'+f_name
-        
-        if (choice_int<7):
-            result+=name_combo+'@'+str(domain)
+
+        domain = choice(self.domain_list)
+
+        name_formats = ["{f}{last}", "{first}{last}",
+                        "{first}.{l}", "{first}_{l}",
+                        "{first}.{last}", "{first}_{last}",
+                        "{last}_{first}", "{last}.{first}"]
+        name_fmt_choice = choice(name_formats)
+        name_combo = name_fmt_choice.format(f=f_name[0], l=l_name[0], first=f_name, last=l_name)
+
+        if choice_int < 7:
+            email = name_combo + '@' + str(domain)
         else:
-            random_int = randint(11,99)
-            result+=name_combo+str(random_int)+'@'+str(domain)
+            random_int = randint(11, 99)
+            email = name_combo + str(random_int) + '@' + str(domain)
 
-        return result
+        return email
 
-    def city_real(self,seed=None):
+    def city_real(self, seed=None):
         '''
         Picks and returns a random entry out of 385 US cities
         seed: Currently not used. Uses seed from the pydb class if chosen by user
         '''
-        import os
-        from six import moves
-        import ssl
-        import random
-        from random import randint,choice
         random.seed(self.seed)
 
-        path = "US_Cities.txt"

```

Comment:
Moved this out into an initialization function so we don't read the file every time we want to generate a __single__ city!